### PR TITLE
Poissonloss

### PR DESCRIPTION
**Pull request by avantikalal**
Added an option to train using poisson loss for regression.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -83,13 +83,20 @@ def get_losses(task, mse_weight, pearson_weight, gpu):
     """
 
     if task == "regression":
-        loss_func = MultiLoss(['mse', 'pearsonloss'], [
-                              mse_weight, pearson_weight], device=gpu)
+        if poisson_weight > 0:
```
Comment (tijyojwad) [inline main.py]: Since there are several conditionals in this function now, wondering if the following re-write makes it more readable -

```
reg_loss_func = MultiLoss('poissonloss', poisson_weight, device=gpu) if poisson_weight > 0 else MultiLoss(['mse', 'pearsonloss'], [mse_weight, pearson_weight], device=gpu)
cla_loss_func = MultiLoss('bce', 1, device=gpu)

if task == "regression":
    loss_func = reg_loss_func
elif task == "classification":
    loss_func = "cla_loss_func
else:
    loss_func = [reg_loss_func, cla_loss_func]

return loss_func
```

File: claragenomics/dl4atac/train/custom_losses.py [hunk 2]
```diff
@@ -21,3 +21,13 @@ def forward(self, input, targets):
...
         return r_loss
+
+
+class PoissonLoss(nn.Module):
+
+    def __init__(self):
+        super(PoissonLoss, self).__init__()
+
+    def forward(self, input, targets):
+        loss = torch.mean(input - targets*torch.log(input + 1e-7))
```
[the remark this answers is not in the document]
Comment (avantikalal) [reply claragenomics/dl4atac/train/custom_losses.py]: done

File: claragenomics/dl4atac/train/dataset.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 import sys
 
 
-MAX_FILES = 20
+MAX_FILES = 26
```
[the remark this answers is not in the document]
Comment (avantikalal) [reply claragenomics/dl4atac/train/dataset.py]: done

File: main.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ def set_random_seed(seed):
         mpu.model_parallel_cuda_manual_seed(seed)
 
 
-def get_losses(task, mse_weight, pearson_weight, gpu):
+def get_losses(task, mse_weight, pearson_weight, gpu, poisson_weight=None):
```
[the remark this answers is not in the document]
Comment (avantikalal) [reply main.py]: fixed

File: main.py (hunk 1 above)
Comment (avantikalal) [reply main.py]: replaced

File: main.py [hunk 5]
```diff
@@ -70,26 +70,30 @@ def set_random_seed(seed):
...
     if task == "regression":
-        loss_func = MultiLoss(['mse', 'pearsonloss'], [
-                              mse_weight, pearson_weight], device=gpu)
+        loss_func = reg_loss_func
     elif task == "classification":
-        loss_func = MultiLoss('bce', 1, device=gpu)
-    elif task == 'both':  # shouldn't reach here for now
-        loss_func = [MultiLoss(['mse', 'pearsonloss'], [mse_weight, pearson_weight], device=gpu),
-                     MultiLoss('bce', 1, device=gpu)]
+        loss_func = "cla_loss_func
```
Comment (tijyojwad) [inline main.py]: whoops, looks like there was a typo in my comment. there's an `"` which needs to be removed

Comment (avantikalal) [reply main.py]: removed

**Comment by tijyojwad** [comment]
rerun tests
